lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -821,6 +866,32 @@ export abstract class CacheManager implements ICacheManager {
         await Promise.all(removedCredentials);
     }
 
+    /**
+     * Replaces the single-tenant account with a new account object that contains the array of tenants.
+     * Returns the updated account entity.
+     * @param accountKey
+     * @param accountEntity
+     * @returns
+     */
+    protected updateOutdatedCachedAccount(
+        accountKey: string,
+        accountEntity: AccountEntity
+    ): AccountEntity {
+        if (accountEntity.isSingleTenant()) {
+            const updatedAccount = Object.assign(new AccountEntity(), {
+                ...accountEntity,
+                tenants: [accountEntity.realm],
+            });
+            this.removeOutdatedAccount(accountKey);
```

Comment:
Do we have to remove then add because the keys won't necessarily be the same? How much work would it be to compare old key to new key and then only remove if they don't match?

Reply:
The main issue is the tenants array not being populated. If you are come from a previous version and have an account with home tenant ID token, then come to multi-tenant and authenticate with a new tenant, the tenants array would include the new tenant but not the home tenant that was already logged in.

I guess a workaround might be to add the home tenant when new tenants are registered if it's not already in the array?

Reply:
Understood but writing to a key will overwrite the value's contents with the new value. If the keys are the same you can just do a write and skip the remove.

Reply:
Not to say the key doesn't matter. The realm in the key is now the home tenant since we only store the home account info now and source tenant-specific data from ID tokens.

 If in a previous version someone logs in with a guest tenant account, their account key will end with -guestTenantId and we now look for -homeTenantId so it may result in cache misses. This would catch those accounts and convert them.

Reply:
I'll add the key comparison and only remove if they don't match